### PR TITLE
Manacher's Algorithm For (Longest Palindromic Substring)#76

### PR DESCRIPTION
**Pull request by singhkunal01**
## Related Issue
- [ADD] Manacher's Algorithm For (Longest Palindromic Substring) #76 

## Proposed Changes
- I have added the short and crisp explanation with comments and code, You can check if something better i can do Let Me Know !!

## Additional Info
- Manacher’s Algorithm helps us find the longest palindromic substring in the given string in better complexity then the Brute Force Solution also the Dynamic Programming Solution in Linear Time and Linear Space.It will reduce the number of expands for the each index it saves the length of previously occured palindrome length.

## Checklist

- [x] ✅ My code follows the code style of this project.
- [ ] 📝 My change requires a change to the documentation.
- [x] 🎀 I have updated the documentation accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
| Screenshot #1      | Screenshot #2  |
| ----------- | ----------- |
| Main Code  | Code with Solution     |
|Manacher1 | Mancher2    |

### CODE REVIEW

**Comment by singhkunal01** [comment]
hey @miraj0507 is it everything fine ?

**Comment by Lakhankumawat** [comment]
1. Delete the unncessary `png` Files 
2. Fetch Upstream And EDIT README of the folder according to your algorithm
3. Uncheck all the checks which do not apply to you

**Comment by Lakhankumawat** [comment]
You need to follow particular guidelines for your README , please follow
these guidelines [click here](https://github.com/Lakhankumawat/LearnCPP#to-add-programs-using-stl-follow-same-approach-as--algorithms) 
also you can see how your README should look like [see here](https://github.com/Lakhankumawat/LearnCPP/blob/main/S-SortingAlgorithms/README.md) 
for more please go through these [here](https://youtu.be/7UgC3HDETRA)

**Comment by singhkunal01** [comment]
Sure i will change all the changes you want @Lakhankumawat give me some time.

File: D-DynamicProgramming/ManachersAlgorithm/ManachersAlgorithm.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
+/*
+ Q. What Does This MANACHER's Algorithm Does ?
+
```
Comment (siddhi-244) [inline D-DynamicProgramming/ManachersAlgorithm/ManachersAlgorithm.cpp:3]: Try to add this explanation in Readme

**Comment by singhkunal01** [comment]
Hi @Lakhankumawat  can you please help me that what are you waiting for ?? Is there any problem in my solution ?

**Review by siddhi-244 — CHANGES_REQUESTED** [review]
can uh explain the  approach a bit more in form of steps ?

**Comment by siddhi-244** [comment]
Closing this pr!

**Pull request closed without merging**